internal/service/alarms/internal/alarms_server.go

```diff
@@ -41,8 +41,21 @@ func (a *AlarmsServer) DeleteSubscription(ctx context.Context, request api.Delet
 }
 
 func (a *AlarmsServer) GetSubscription(ctx context.Context, request api.GetSubscriptionRequestObject) (api.GetSubscriptionResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	subsModel, err := a.AlarmsRepository.GetAlarmSubscriptionWithUuid(ctx, request.AlarmSubscriptionId)
+	if errors.Is(err, utils.ErrNotFound) {
+		// Nothing found
+		return api.GetSubscription404ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"UUID": request.AlarmSubscriptionId.String(),
```

Comment:
nit:  since our yaml defines this variable as `subscriptionId` which should refer to it by name here rather than "UUID"